### PR TITLE
Location_finder file added successfully

### PR DESCRIPTION
**Pull request by SinghHimadri**
## Issue that this pull request solves
#31 
Closes: # (issue number)

## Proposed changes

Brief description of what is fixed or changed
Currently, the location can be found using number since it requires to enter the number in international format eg:+91888888888 with country code.This can be executes using -wl command then enter the number in valid format.Since format we save our phone numbers is without adding country in future I'll be adding a option to choose a contact or to enter the number manually.This is working successfully It contains some additional information too like Carrier and time zone etc.
## Types of changes

_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update (Documentation content changed)
- [ ] Other (please describe): 

## Checklist

_Put an `x` in the boxes that apply_

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

## Screenshots

Please attach the screenshots of the changes made in case of change in user interface
Screenshot from 2020-04-24 15-00-38

## Other information

Any other information that is important to this pull request
I would request you to not close this issue as I'll be adding option in this feature.

### CODE REVIEW

File: wplay/location_finder.py [hunk 1]
```diff
@@ -1,8 +1,102 @@
...
+    global numberCountryCode
+    global numberCountry
+
+    print(code_info + 'Running local scan...')
+
+    FormattedPhoneNumber = "+" + formatNumber(InputNumber)
+
+    try:
+        PhoneNumberObject = phonenumbers.parse(FormattedPhoneNumber, None)
+    except:
```
Comment (rpotter12) [inline wplay/location_finder.py:46]: Codacy Issue found: [No exception type(s) specified](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5414447)

**Comment by rpotter12** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- wplay/location_finder.py  19
- wplay/__main__.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5414447&bid=17752110)

File: wplay/__main__.py [hunk 2]
```diff
@@ -171,9 +161,18 @@ async def get_and_match_args(parser):
     elif args.get_profile_photos:
         await get_media.get_profile_photos()
 
-    # elif args.wlocation:
-    #     loactionfinder.finder(args.target)
-
+    elif args.location_finder:
+        await location_finder.location_finder()
+
```
Comment (rpotter12) [inline wplay/__main__.py:166]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5414447)

**Comment by rpotter12** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- wplay/location_finder.py  19
- wplay/__main__.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5414447&bid=17752110)

File: wplay/location_finder.py (hunk 1 above)
Comment (rpotter12) [inline wplay/location_finder.py:46]: Codacy Issue found: [No exception type(s) specified](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5414447)

File: wplay/__main__.py [hunk 3]
```diff
@@ -148,17 +148,7 @@ async def get_and_match_args(parser):
...
-        if args.target is None:
-            parser.print_help()
-            parser.exit()
-        try:
-            bID: int = int(sys.argv[3])
-        except (IndexError, ValueError):
-            bID: int = 0
-        save_chat.runMain('pull', str(args.target), bID)
-
+
```
Comment (rpotter12) [inline wplay/__main__.py:151]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5414447)

**Comment by rpotter12** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- wplay/location_finder.py  19
- wplay/__main__.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5414447&bid=17752110)

File: wplay/location_finder.py (hunk 1 above)
Comment (rpotter12) [inline wplay/location_finder.py:46]: Codacy Issue found: [No exception type(s) specified](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5414447)

**Comment by rpotter12** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- wplay/location_finder.py  19
- wplay/__main__.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5414447&bid=17752110)

File: wplay/location_finder.py (hunk 1 above)
Comment (rpotter12) [inline wplay/location_finder.py:46]: Codacy Issue found: [Try, Except, Pass detected.](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5414447)

Comment (rpotter12) [inline wplay/location_finder.py:46]: Codacy Issue found: [No exception type(s) specified](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5414447)

**Comment by rpotter12** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- wplay/location_finder.py  3
- wplay/__main__.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5414447&bid=17752110)

File: wplay/location_finder.py [hunk 4]
```diff
@@ -1,8 +1,100 @@
...
+    global localNumber
+    global internationalNumber
+    global numberCountryCode
+    global numberCountry
+
+    print(code_info + 'Running local scan...')
+
+    FormattedPhoneNumber = "+" + formatNumber(InputNumber)
+
+
```
Comment (rpotter12) [inline wplay/location_finder.py:44]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5414447)

**Comment by rpotter12** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- wplay/location_finder.py  3
- wplay/__main__.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5414447&bid=17752110)

**Comment by SinghHimadri** [comment]
@rpotter12 @xandao6 @priyansh19 Please review my PR.

**Comment by priyansh19** [comment]
There are some issues in your PR please resolve file complexity

**Comment by rpotter12** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- wplay/location_finder.py  3
- wplay/__main__.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5414447&bid=17752110)

File: wplay/location_finder.py [hunk 5]
```diff
@@ -1,8 +1,97 @@
...
+def scanNumber(InputNumber):
+    print(code_title + "[!] ---- Fetching informations for {} ---- [!]".format(formatNumber(InputNumber)))
+
+    localScan(InputNumber)
+
+    global number
+    global localNumber
+    global internationalNumber
+    global numberCountryCode
+    global numberCountry
```
Comment (xandao6) [inline wplay/location_finder.py:81]: Do not use global variables please, is not a good practice.

File: wplay/location_finder.py [hunk 6]
```diff
@@ -1,8 +1,97 @@
...
+# region LOGGER
+__logger = Logger(Path(__file__).name)
+# endregion
 
-def location_finder(name):
+code_info = '[*] '
+code_warning = '(!) '
+code_result = '[+] '
+code_error = '[!] '
+code_title = ''
```
Comment (xandao6) [inline wplay/location_finder.py:22]: Maybe a dictionary will fit better

**Comment by rpotter12** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- wplay/location_finder.py  15
- wplay/__main__.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5414447&bid=17752110)

**Comment by rpotter12** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- wplay/location_finder.py  15
- wplay/__main__.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5414447&bid=17752110)

File: wplay/target_info.py [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+    phone_number = input("Enter full number with country code e.g:'+91888888888':")
+    scanNumber(phone_number)
+    """
+    # to find location by ip address
+    print('Get you ipinfo token from https://ipinfo.io/account')
+    ip_address = '*'
+    token = str(input("Enter your ipinfo token: "))
+    ip_string = 'curl ipinfo.io/'+ip_address+'?token='+token+''
+    os.system(ip_string)
+    """
```
Comment (rpotter12) [inline wplay/target_info.py:106]: Codacy Issue found: [String statement has no effect](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5414447)

**Comment by rpotter12** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- wplay/target_info.py  16
- wplay/__main__.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5414447&bid=17752110)

File: wplay/target_info.py (hunk 7 above)
Comment (SinghHimadri) [reply wplay/target_info.py:106]: This statement wasalready present in initial file which was added by you.

**Comment by SinghHimadri** [comment]
> @SinghHimadri 2 things:
> 
> 1. your code should follow PEP8 rule.
>    Install pycodestyle by `pip install pycodestyle`
>    Then check your changed file by `pycodestyle --first <filename>`
>    Eg: `pycodestyle --first location_finder.py`
>    By this you will get all the lines numbers where you are not following PEP8 rule
> 2. these code will be more good for target info. So change the file name and arguments according to target_info.

I have done all the changes as directed.

**Comment by rpotter12** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- wplay/__main__.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5414447&bid=17752110)

**Comment by SinghHimadri** [comment]
@rpotter12 @xandao6 I have done all the changes as asked please review.

**Comment by SinghHimadri** [comment]
> There are some issues in your PR please resolve file complexity

I have resolved this please see.

File: wplay/target_info.py [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+    __logger.info("Writing target's information")
+    phone_number = input("Enter full number with country code e.g:'+91888888888':")
+    scanNumber(phone_number)
+    """
+    # to find location by ip address
+    print('Get you ipinfo token from https://ipinfo.io/account')
+    ip_address = '*'
+    token = str(input("Enter your ipinfo token: "))
+    ip_string = 'curl ipinfo.io/'+ip_address+'?token='+token+''
+    os.system(ip_string)"""
```
Comment (rpotter12) [inline wplay/target_info.py:105]: Codacy Issue found: [String statement has no effect](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5414447)

**Comment by rpotter12** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- wplay/target_info.py  16
- wplay/__main__.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5414447&bid=17752110)

**Comment by rpotter12** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- wplay/target_info.py  16
- wplay/__main__.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5414447&bid=17752110)

**Pull request merged**